### PR TITLE
WT-2026 Allow forced eviction to split pages when BTREE_NO_EVICTION set

### PR DESCRIPTION
**Pull request by keithbostic**
@sulabhM, for your review/consideration.

### CODE REVIEW

File: src/evict/evict_page.c [hunk 1]
```diff
@@ -459,14 +462,22 @@ __evict_review(
...
 		 */
 		if (LF_ISSET(WT_EVICT_INMEM_SPLIT))
 			return (__wt_split_insert(session, ref));
+
+		/*
+		 * If we can't split the page, check if eviction is possible
+		 * for this file (eviction is turned off during some parts of
+		 * checkpoint).
+		 */
+		if (F_ISSET(btree, WT_BTREE_NO_EVICTION))
```
Comment (agorrod) [inline src/evict/evict_page.c]: @keithbostic The problem with this change is that the current code expects the ``WT_BTREE_NO_EVICTION`` flag to be ignored here when closing a file - @sulabhM is that correct?

Comment (sulabhM) [inline src/evict/evict_page.c]: @agorrod : That is correct. While closing a file we have WT_BTREE_NO_EVICTION set and we allow the reconciliation to happen. @keithbostic : If we return error from this point if WT_BTREE_NO_EVICTION is set, we will unintentionally fail for file-close case as well.

Comment (keithbostic) [inline src/evict/evict_page.c]: @sulabhM we have the `closing` flag, is checking it here sufficient?

**Comment by agorrod** [comment]
I'm going to close this branch - I prefer the other solution, and think there is less risk.

**Pull request closed without merging**